diff --git a/src/unix/linux.c b/src/unix/linux.c
index 48b9c2c4..4ae67296 100644
--- a/src/unix/linux.c
+++ b/src/unix/linux.c
@@ -114,7 +114,7 @@
 # endif
 #endif /* __NR_getrandom */
 
-#define HAVE_IFADDRS_H 1
+#undef HAVE_IFADDRS_H
 
 # if defined(__ANDROID_API__) && __ANDROID_API__ < 24
 # undef HAVE_IFADDRS_H
diff --git a/test/test-list.h b/test/test-list.h
index 78ff9c2d..c05ab80b 100644
--- a/test/test-list.h
+++ b/test/test-list.h
@@ -483,7 +483,7 @@ TEST_DECLARE   (poll_nested_kqueue)
 TEST_DECLARE   (poll_multiple_handles)
 
 TEST_DECLARE   (ip4_addr)
-TEST_DECLARE   (ip6_addr_link_local)
+// NO_TEST_DECLARE   (ip6_addr_link_local)
 TEST_DECLARE   (ip_name)
 
 TEST_DECLARE   (poll_close_doesnt_corrupt_stack)
@@ -1157,7 +1157,7 @@ TASK_LIST_START
   TEST_ENTRY  (thread_affinity)
   TEST_ENTRY  (dlerror)
   TEST_ENTRY  (ip4_addr)
-  TEST_ENTRY  (ip6_addr_link_local)
+//  NO_TEST_ENTRY  (ip6_addr_link_local)
   TEST_ENTRY  (ip_name)
 
   TEST_ENTRY  (queue_foreach_delete)
diff --git a/test/test-platform-output.c b/test/test-platform-output.c
index 5839f52d..8ba16d1f 100644
--- a/test/test-platform-output.c
+++ b/test/test-platform-output.c
@@ -112,6 +112,7 @@ TEST_IMPL(platform_output) {
 #endif
   uv_free_cpu_info(cpus, count);
 
+#if 0
   err = uv_interface_addresses(&interfaces, &count);
   ASSERT(err == 0);
 
@@ -147,6 +148,7 @@ TEST_IMPL(platform_output) {
     }
   }
   uv_free_interface_addresses(interfaces, count);
+#endif
 
   err = uv_os_get_passwd(&pwd);
   ASSERT_EQ(err, 0);
